### PR TITLE
Add `target_build_settings` Starlark integreation

### PR DESCRIPTION
**Pull request by brentleyjones**

### CODE REVIEW

**Review by thii — APPROVED** [review]

File: test/internal/pbxproj_partials/write_target_build_settings_tests.bzl [hunk 1]
```diff
@@ -0,0 +1,796 @@
...
+    )
+
+    if ctx.attr.expect_debug_settings:
+        expected_debug_settings = _DEBUG_SETTINGS_DECLARED_FILE
+        expected_declared_files[expected_debug_settings] = None
+        expected_inputs = swift_debug_settings_to_merge
+        expected_outputs.append(expected_debug_settings)
+    else:
+        expected_debug_settings = None
+        expected_inputs = []
```
Comment (mattrobmattrob) [inline test/internal/pbxproj_partials/write_target_build_settings_tests.bzl:51]: [nit] Just to match the `expected_outputs` pattern above:
```suggestion
    expected_inputs = []

    if ctx.attr.expect_debug_settings:
        expected_debug_settings = _DEBUG_SETTINGS_DECLARED_FILE
        expected_declared_files[expected_debug_settings] = None
        expected_inputs = swift_debug_settings_to_merge
        expected_outputs.append(expected_debug_settings)
    else:
        expected_debug_settings = None
```

File: xcodeproj/internal/pbxproj_partials.bzl [hunk 2]
```diff
@@ -579,8 +602,228 @@ def _write_pbxtargetdependencies(
...
+        *   A `File` containing Swift debug settings for the target, or `None`.
+        *   A `list` of `File`s containing C or C++ compiler arguments. These
+            files should be added to compile outputs groups to ensure that Xcode
+            has them available for the `Create Compile Dependencies` build
+            phase.
+    """
+    generate_swift_debug_settings = swift_args or is_top_level_target
+
+    if not (generate_build_settings or generate_swift_debug_settings):
+        return None, None, []
```
Comment (mattrobmattrob) [inline xcodeproj/internal/pbxproj_partials.bzl:682]:
```suggestion
        return None, None, EMPTY_LIST
```

File: xcodeproj/internal/pbxproj_partials.bzl [hunk 3]
```diff
@@ -579,8 +602,228 @@ def _write_pbxtargetdependencies(
...
+    Args:
+        actions: `ctx.actions`.
+        apple_generate_dsym: `cpp_fragment.apple_generate_dsym`.
+        certificate_name: The name of the certificate to use for code signing.
+        colorize: A `bool` indicating whether to colorize the output.
+        conly_args: A `list` of `Args` for the C compile action for this target.
+        cxx_args: A `list` of `Args` for the C++ compile action for this target.
+        device_family: A value as returned by `get_targeted_device_family`.
+        entitlements: An optional entitlements `File`.
+        extension_safe: If `True, `APPLICATION_EXTENSION_API_ONLY` will be set.
```
Comment (mattrobmattrob) [inline xcodeproj/internal/pbxproj_partials.bzl:641]:
```suggestion
        extension_safe: If `True`, `APPLICATION_EXTENSION_API_ONLY` will be set.
```

**Review by mattrobmattrob — APPROVED** [review]

File: test/internal/pbxproj_partials/write_target_build_settings_tests.bzl (hunk 1 above)
Comment (brentleyjones) [reply test/internal/pbxproj_partials/write_target_build_settings_tests.bzl:51]: I like my way in performance critical code. I'll go with your way for readability.

**Pull request merged**